### PR TITLE
feat(client|redux|react): implement and both orders and returns area to integrate the orders split experience and multiple returns flow.

### PR DESCRIPTION
**Pull request by pedro-gbf**
## Description

This implements all the needed changes to support the new order split flow while supporting
the multiple returns flow so both orders and returns area were reviewed in the following
changes lists:
- Add User Returns clients and redux actions.
- Add the Order change address clients and redux actions.
- Improved jsdoc of both areas.
- Removed unnecessary async declarations in fixtures.
- Renamed some types to be consistent with naming convetions.
- Removed unnecessary parameters to `useOrderReturnOptions` hook.
- Improved unit tests of the reducers and selectors of both areas.

BREAKING CHANGES:
- Removed the entities mapper specific fetch order logic to mergeinformation since we no longer have merging issues.
- `getOrder` was reviewed
- `getOrderReturns` was migrated and reviewed on the returns area.
- `getOrderMerchants` was deleted and a new selector was born to replace it called
 `getOrderSummaries` that returns all the order summaries of a certain order id.
- `getOrderItemsByMerchant` was renamed to `getOrderItemsBySummary` and it now returns
 the order items of an order summary instead of splitting them by merchant.
- `getOrderItemQuantity` was reviewed to remove an unnecessary restriction
- `getUserOrder` is now normalising the date
- `getOrderItems` is now normalising the result
- `getOrderItem` is now normalising the result
- Tests were adapted and added to test the mentioned changes
- Removed the reset of the return options since they are already resetted when a
 new user is fetches or when a user logins. There is no use case to manually reset
 this data for a user.
-  Revamped the `returnOption` schema to wrap around the entire object that contains
 sub `returnOptions` per type. The older schematisation implied that there was a
need to reference each of the sub returnOption and there is none. Also the
identifier of the this entity has been changed to be the `merchantOrderId` property
 so we can link it to the order items if needed.
- Removed the artificial merchant normalization.
- `getOrderReturnOptions` selector now returns a list of return options denormalized
 organised by `merchantOrderId` this id can be linked to order items.
- `getReturnOption`  is now normalising the result. You now need to provide a
`merchantOrderId` to fetch it.
- `fetchOrderReturnOptions` now saves the array with the merchantOrderIds identifying
 the returnOptions entities in the Orders area of the redux store.
- Removed `getGuestOrder` client and `fetchGuestOrder` action since it works the same way as `getOrder` and `fetchOrder`.
- `getOrderItemQuantity` was replaced by `getOrderItemProductQuantity`, and now returns the quantity of a product in a given order.

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by andreecosta — APPROVED** [review]

File: packages/client/src/orders/__tests__/getUserOrders.test.ts [hunk 1]
```diff
@@ -24,6 +25,21 @@ describe('getUserOrders', () => {
     );
   });
 
+  it('should handle a client request for multiple pages', async () => {
+    mswServer.use(fixtures.success(mockOrdersResponse2));
+
+    // await expect(getUserOrders(userId)).resolves.toEqual(mockOrdersResponse2);
```
[the remark this answers is not in the document]
Comment (pedro-gbf) [reply packages/client/src/orders/__tests__/getUserOrders.test.ts]: Removed it 👍

File: packages/redux/src/orders/reducer.ts [hunk 2]
```diff
@@ -328,63 +115,17 @@ export const entitiesMapper = {
...
-              orderReturns.forEach(returnId => {
-                delete draftState.returns?.[returnId];
-              });
-            }
-
-            delete merchantOrder.returns;
-          });
-        }
-      });
+      const returnOptions = state.returnOptions;
```
Comment (boliveira) [inline packages/redux/src/orders/reducer.ts:118]:
```suggestion
      const returnOptions = draftState.returnOptions;
```

File: packages/redux/src/orders/reducer.ts [hunk 3]
```diff
@@ -328,63 +115,17 @@ export const entitiesMapper = {
...
+
+      if (returnOptions) {
+        Object.values(returnOptions).forEach(({ merchantOrderId, orderId }) => {
+          if (orderIdsToEnumerate.includes(orderId)) {
+            delete draftState.returnOptions?.[merchantOrderId];
```
Comment (boliveira) [inline packages/redux/src/orders/reducer.ts:123]:
```suggestion
            delete returnOptions?.[merchantOrderId];
```

File: packages/redux/src/entities/types/returnOptions.types.ts [hunk 4]
```diff
@@ -1,25 +1,16 @@
...
+export type ReturnOptionEntity = MerchantOrderReturnOptions & {
+  orderId: Order['id'];
 };
-
-export type ReturnOptionEntityDenormalized = Omit<
-  ReturnOptionEntity,
-  'merchant'
-> & {
-  merchant: MerchantEntity;
+export type ReturnOptionEntityDenormalized = ReturnOptionEntity & {
```
Comment (boliveira) [inline packages/redux/src/entities/types/returnOptions.types.ts:10]: Add an empty line here for spacing, please.

File: packages/client/src/orders/types/postOrderShippingAddressChangeRequest.types.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import type { Config } from '../../types/index.js';
+import type { Order, OrderShippingAddressChangeRequest } from './index.js';
+
+export type PostOrderShippingAddressChangeRequest = (
+  orderId: Order['id'],
+  data: OrderShippingAddressChangeRequestData,
+  config?: Config,
+) => Promise<number>;
+
+export type OrderShippingAddressChangeRequestData = Pick<
```
Comment (boliveira) [inline packages/client/src/orders/types/postOrderShippingAddressChangeRequest.types.ts:10]: Sorry, missed this one. Can you rename this to `PostOrderShippingAddressChangeRequestData` instead to be consistent, please?

Comment (pedro-gbf) [reply packages/client/src/orders/types/postOrderShippingAddressChangeRequest.types.ts:10]: Done 👍

File: packages/redux/src/orders/reducer.ts (hunk 3 above)
Comment (pedro-gbf) [reply packages/redux/src/orders/reducer.ts]: Done 👍

**Review by jpklzm — APPROVED** [review]

**Pull request merged**